### DISCUSSION TITLE
Does intel pcm capture kernel activity?

### DISCUSSION THREAD

**Discussion by anonymous1782** [opcm/pcm#532] [Q&A]
Hi, 

I wonder if the intel pcm tool can capture linux kernel activity.

Colud anyone let me know how pcm measure memory bandwidth?

Thanks

**Comment by rdementi** [comment]
the memory bandwidth is measured using the uncore performance monitoring units: https://www.intel.com/content/www/us/en/developer/articles/technical/intel-sdm.html#inpage-nav-7
and some bandwidth metrics use Memory Bandwidth Monitoring (MBM) technology:  https://www.intel.com/content/www/us/en/developer/articles/technical/introduction-to-memory-bandwidth-monitoring.html

Intel PCM captures all activity, not specifically Linux kernel activity
